src/openrct2/actions/RideSetColourScheme.hpp

```diff
@@ -0,0 +1,73 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2018 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../Cheats.h"
+#include "../common.h"
+#include "../core/MemoryStream.h"
+#include "../interface/Window.h"
+#include "../localisation/Localisation.h"
+#include "../localisation/StringIds.h"
+#include "../management/Finance.h"
+#include "../ride/Ride.h"
+#include "../world/Park.h"
+#include "../world/Sprite.h"
+#include "GameAction.h"
+
+struct RideSetColourSchemeAction : public GameActionBase<GAME_COMMAND_SET_COLOUR_SCHEME, GameActionResult>
+{
+private:
+    int32_t _x, _y, _z, _direction, _trackType;
+    uint16_t _newColourScheme;
```

Comment:
All of these fields should be `const`

Reply:
Ugh, we can't make them const because https://github.com/OpenRCT2/OpenRCT2/pull/8138 needs them the way they are. I'll see if I can come up with a change to address that at some later point.